src/utils/peer.ts

```diff
@@ -29,63 +24,92 @@ function sendPlayerName(c: DataConnection) {
   });
 }
 
-export function boardcastPlayerList() {
-  if (!isHost.value) return;
-  const playerIdAndNamePairs = [...playerMap.value];
-  boardcastMessage(() => ({
-    type: MessageType.PLAYER_LIST,
-    value: playerIdAndNamePairs,
-  }));
-}
-
 effect(() => {
+  if (!peer.value) return;
+  addListenersToPeer();
+});
+
+function addListenersToPeer() {
   const p = peer.value;
   if (!p) return;
-  p.off("error")
-    .on("error", (e) => handleErrors(e))
-    .off("connection")
-    .on("connection", (c) => handleConnectionToTheHost(c))
-    .off("open")
-    .on("open", () => handleNonHostPeerOpened(p));
-});
+  p.on("error", (e) => handlePeerError(e))
+    .on("connection", (c) => handleReceivedPeerConnection(c))
+    .on("disconnected", () => handlePeerDisconnectedFromTheServer(p))
+    .on("open", () => handlePeerOpened(p));
+}
 
-function handleConnectionToTheHost(c: DataConnection) {
+function handleReceivedPeerConnection(c: DataConnection) {
   if (!isHost.value) return;
-  applyMessageHandler(c);
-  connectionMap.value = new Map([...connectionMap.value, [c.peer, c]]);
-  c.off("open").on("open", () => {
-    console.info(`New player ${c.peer} joined.`);
-    sendPlayerName(c);
-    boardcastPlayerList();
-  });
+  handleReceivedPeerConnectionToTheHost(c);
 }
 
-function handleNonHostPeerOpened(p: Peer) {
+function handlePeerDisconnectedFromTheServer(p: Peer) {
```

Comment:
What is `server`?

Reply:
This is the signal server (or consider it as a helper server). However, in this project's scope, this is the only "server" concept we are currently using.